src/webapp/src/client/app/lotteries/lotteries-widget/lotteries-widget.directive.js

```diff
@@ -0,0 +1,134 @@
+(function(){
+    'use strict';
+
+    angular
+        .module('simoonaApp.Lotteries')
+        .constant("lotteryStatus", {
+            Drafted: 1,
+            Started: 2,
+            Aborted: 3,
+            Ended: 4
+        })
+        .directive('aceLotteriesWallWidget', lotteriesWallWidget)
+        .filter('timespan', function () {
+            return getTimespan;
+        });
+
+        lotteriesWallWidget.$inject = [
+            'authService',
+            'lotteryRepository',
+            'lotteryStatus',
+            'localeSrv',
+            '$window'
+        ];
+
+        function lotteriesWallWidget(authService, lotteryRepository, lotteryStatus, localeSrv, $window){
+            var directive = {
+                restrict: 'E',
+                templateUrl: 'app/lotteries/lotteries-widget/lotteries-widget.html',
+                link: linkFunc
+            };
+            return directive;
+
+            function linkFunc(scope) {
+                if(!$window.lotteriesEnabled)
+                {
+                    return;
+                }
+                scope.lotteryStatus = lotteryStatus;
+                scope.latestLotteries = [];
+                scope.getRemainingTime = getRemainingTime;
+                scope.hasLotteryPermisions = authService.hasPermissions(['LOTTERY_BASIC']);
+
+                getLotteryWidgetInfo();
+
+                function getLotteryWidgetInfo(){
+                    lotteryRepository.getLotteryWidgetInfo().then(function(result) {
+                        scope.latestLotteries = result;
+                        filterEndedLotteries();
+                    })
+                }
+                function filterEndedLotteries(){
+                    angular.forEach(scope.latestLotteries, (lottery) => {
+                        if(moment.utc(lottery.endDate)-moment() < 0)
+                        {
+                            scope.latestLotteries.remove(lottery);
+                        }
+                    });
+                }
+            }
+            function getRemainingTime(input){
```

Comment:
This part of the code is questionable. To return the remaining time is fairly difficult due to different languages (moment-locale package is not installed), so i wrote a helper function which returns translated values. Basically this function returns only unit of time (ex: tomorrow) with no time value or unit of time with time value (ex: 2 days, 3 weeks, etc..). So if anyone has a better idea how to implement remaining time with different languages, I'll be happy to refactor this code.